nix/checks.nix

```diff
@@ -9,16 +9,44 @@
   }: {
     checks =
       {
+        black =
+          pkgs.runCommandLocal "black" {
+            nativeBuildInputs = [pkgs.python3.pkgs.black];
+          } ''
+            cd ${../.}
```

Comment:
@brianmcgee We're using this quite frequently - it'd probably make sense to construct a gitingore-cleaned src once in a let binding and then just pass it down to all checks (or even doing the filtering we currently do at runtime in Nix), hopefully resulting in less rebuilds.

Reply:
@flokli You can enable black in treefmt and remove this. There's already a treefmt check registered.

Reply:
pushed your commit to my branch, thanks!